examples/meta/generator/translate.py

```diff
@@ -458,6 +461,42 @@ def translateStatement(self, statement):
         template = Template(self.targetDict["Statement"])
         return template.substitute(statement=translation)
 
+    def translateListInit(self, init):
+        """ Translate initialisation statement AST
+        Args:
+            listinit: [typeAST, identifierAST, argumentListAST]
+        """
+        typeDict = init[0]
+        typeString = self.translateType(typeDict)
+        nameString = init[1]["Identifier"]
+        initialisation = init[2]
+        typeKey = list(typeDict.keys())[0]
+
+        self.variableTypes[nameString] = typeDict
+
+        if list(initialisation.keys())[0] == "ArgumentList":
+            template = Template(self.targetDict["Init"]["Construct"])
+            # custom SGType construction
+            if typeKey == "ShogunSGType"\
+                and init[0][typeKey] in self.targetDict["ListInit"]:
+                template = Template(self.targetDict["ListInit"][init[0][typeKey]])
```

Comment:
This should probably be moved out of the condition. This substitution only works if there is an implementation of the list initialiser of the target language